test/windows/WsladiagTests.cpp

```diff
@@ -0,0 +1,159 @@
+    {
+        ValidateWslaDiagOutput(L"", 0, L"Usage:");
+    }
+
+    // Test that -h and --help flags produce identical output
+    TEST_METHOD(Help_ShortAndLongFlags_Match)
+    {
+        auto [outH, errH, codeH] = RunWsladiag(L"-h");
+        auto [outLong, errLong, codeLong] = RunWsladiag(L"--help");
+
+        VERIFY_ARE_EQUAL(0, codeH);
+        VERIFY_ARE_EQUAL(0, codeLong);
+
+        VERIFY_ARE_EQUAL(L"", outH);
+        VERIFY_ARE_EQUAL(L"", outLong);
+
+        VERIFY_ARE_EQUAL(errH, errLong);
+        ValidateUsage(errH);
+    }
+
+    // Test that unknown commands show error message and usage
+    TEST_METHOD(UnknownCommand_ShowsError)
+    {
+        auto [out, err, code] = RunWsladiag(L"blah");
+        VERIFY_ARE_EQUAL(1, code);
+
+        const std::wstring combined = out + err;
+        VERIFY_IS_TRUE(combined.find(L"Unknown command: 'blah'") != std::wstring::npos);
+        VERIFY_IS_TRUE(combined.find(L"Usage:") != std::wstring::npos);
+    }
+
+    // Test that shell command without session name shows error
+    TEST_METHOD(Shell_MissingName_ShowsError)
+    {
+        ValidateWslaDiagFailsWith(L"shell", L"The parameter is incorrect.");
+    }
+
+    // Test shell command with invalid session name (silent mode)
+    TEST_METHOD(Shell_InvalidSessionName_Silent)
+    {
+        ValidateWslaDiagFailsWith(L"shell DefinitelyNotARealSession", L"Session not found: 'DefinitelyNotARealSession'");
+    }
+
+    // Test shell command with invalid session name (verbose mode)
+    TEST_METHOD(Shell_InvalidSessionName_Verbose)
+    {
+        const std::wstring name = L"DefinitelyNotARealSession";
+        auto [out, err, code] = RunWsladiag(std::format(L"shell {} --verbose", name));
+        VERIFY_ARE_NOT_EQUAL(0, code);
+
+        VERIFY_IS_TRUE(out.find(std::format(L"[diag] shell='{}'", name)) != std::wstring::npos);
+        VERIFY_IS_TRUE(err.find(L"Session not found") != std::wstring::npos);
+    }
+
+    // Build command line for wsladiag.exe with given arguments
+    static std::wstring BuildWsladiagCmd(const std::wstring& args)
+    {
+        const auto msiPathOpt = wsl::windows::common::wslutil::GetMsiPackagePath();
+        VERIFY_IS_TRUE(msiPathOpt.has_value());
+
+        const auto exePath = std::filesystem::path(*msiPathOpt) / L"wsladiag.exe";
+        const auto exe = exePath.wstring();
+
+        return args.empty() ? std::format(L"\"{}\"", exe) : std::format(L"\"{}\" {}", exe, args);
+    }
+
+    // Execute wsladiag with given arguments and return output, error, and exit code
+    static std::tuple<std::wstring, std::wstring, int> RunWsladiag(const std::wstring& args)
+    {
+        auto cmd = BuildWsladiagCmd(args);
+        return LxsstuLaunchCommandAndCaptureOutputWithResult(cmd.data());
+    }
+
+    static void ValidateWslaDiagOutput(const std::wstring& cmd, const std::wstring& expectedSubstring)
+    {
+        auto [out, err, code] = RunWsladiag(cmd);
+        const std::wstring combined = out + err;
```

Comment:
Same here, I'd recommend having two parameters, one for each output so we can validate them separately. 

I'll also recommend validating the entire strings, that way we'll catch if anything unexpected is printed